odoo/addons/base/models/res_currency.py

```diff
@@ -260,6 +255,15 @@ def is_zero(self, amount):
         self.ensure_one()
         return tools.float_is_zero(amount, precision_rounding=self.rounding)
 
+    @ormcache()
+    @api.model
+    def get_all_currencies(self):
+        currencies = self.search_fetch([], ['symbol', 'position', 'decimal_places'])
```

Comment:
(see my first comment)

IMO, we should sudo() because it shouldn't depends on the user security that call the method the first time (the cache is shared with all the others ones). Also explicit filter out inactive one to avoid depending on the active_test context.
```suggestion
        currencies = self.sudo().search_fetch([('active', '=', True)], ['symbol', 'position', 'decimal_places'])
```
IMO, the sudo() is legit because currency information shouldn't be sensitive.

I know that it isn't coming from your change but it seems the good time to make in more robust, don't you think ?